setup.py

```diff
@@ -269,6 +269,7 @@ def run(self):
     deps["filelock"],  # filesystem locks, e.g., to prevent parallel downloads
     deps["numpy"],
     deps["packaging"],  # utilities from PyPA to e.g., compare versions
+    deps["psutil"],  # memory tracking
```

Comment:
Usually not too keen on adding a new dependency to the core dependencies. Staying slim is a good way to prevent conflicts and to not break everything when there is an issue with another library; so we usually prefer adding soft dependencies to hard dependencies.

Is there something here that would prevent adding `psutil` as a soft dependency, rather than a hard one?

This case seems pretty narrow, as users that don't use the trainer (e.g., only use it for inference, or training with their custom loop/other libraries, only use pipelines, etc) don't need to have that in their system. 

This is how the hard dependencies are defined, with historical exceptions for regex and sacremoses, which could eventually be removed from the hard dependencies. Sacremoses is here because it's necessary to use for the XLM tokenizer, which, unfortunately, will never have a fast tokenizer.

Reply:
Do you mean to add the check inside the metrics the first time it's used? Sure, that would work.

As I covered in OP I tried to find an alternative that is based on a built-in, but surprisingly didn't succeed.